crates/bevy_hierarchy/src/lib.rs

```diff
@@ -1,8 +1,49 @@
 #![warn(missing_docs)]
-//! `bevy_hierarchy` can be used to define hierarchies of entities.
+//! Parent-child relationships for Bevy entities.
 //!
-//! Most commonly, these hierarchies are used for inheriting `Transform` values
-//! from the [`Parent`] to its [`Children`].
+//! You should use the tools in this crate
+//! whenever you want to organize your entities in a hierarchical fashion,
+//! to make groups of entities more manageable,
+//! or to propagate properties throughout the entity hierarchy.
+//!
+//! This crate introduces various tools, including a [plugin]
+//! for managing parent-child relationships between entities.
+//! It provides two components, [`Parent`] and [`Children`],
+//! to store references to related entities.
+//! It also provides [command] and [world] API extensions
+//! to set and clear those relationships.
+//!
+//! More advanced users may also appreciate
+//! [query extension methods] to traverse hierarchies,
+//! and [events] to notify hiearchical changes.
+//! There is also a [diagnostic plugin] to validate property propagation.
+//!
+//! # Hierarchy management
+//!
+//! The methods defined in this crate fully manage
+//! the components responsible for defining the entity hierarchy.
+//! Mutating these components manually may result in hierarchy invalidation.
+//!
+//! Hierarchical relationships are always managed symmetrically.
+//! For example, assigning a child to an entity
+//! will always set the parent in the other,
+//! and vice versa.
+//! Similarly, unassigning a child in the parent
+//! will always unassign the parent in the child.
+//!
+//! ## Despawning entities
+//!
+//! The commands and methods provided by `bevy_ecs` to despawn entities
+//! are not capable of despawning hierarchies of entities, invalidating them.
```

Comment:
I think you meant to say here that using the commands and methods provided by `bevy_ecs` to despawn entities can invalidate the hierarchy. Because if you despawn a hierarchy of entities there is nothing to invalidate

Reply:
I was saying that `bevy_ecs` items are just capable of despawning single entities. If you do that to a single entity that is part of a hierarchy, it surely gets invalidated. You'd have to manually despawn the entire hierarchy, or an entire subtree and then update the hierarchy components on the parent of the root.

I can slightly reword it to make it more clear.